### PR TITLE
feature-9-review

### PR DESCRIPTION
**Pull request by claregillis**
For this feature, I attempted to follow clean architecture but my logic might be a bit off, sorry. I added addReview() to User so that you can add a review to their list. I finished ReviewForm and added all of the necessary classes/documentation for writing a review. You'll see that here and in feature-6-login there are a couple of identical classes (GetUser, UserExists) that I think we can combine later, as long as it doesn't interfere with single-responsibility. 

So far, ReviewDatabaseController just has a dummy method that returns a set user for 2 specific usernames and an empty user otherwise; that's just for testing. There's a TODO to implement that method later where it will return the user from the database or an empty user.

LMK if you have any questions/comments/concerns

### CODE REVIEW

File: src/main/java/entities/User.java [hunk 1]
```diff
@@ -97,12 +98,18 @@ public void addCart() {
...
-    public void writeReview() {
+    public void writeReview(User reviewed, int rating) {
+        new ReviewCreator().createReview(this, reviewed, rating);
 
     }
     public void removeReview() {
 
     }
+
+    //TODO: check if this is necessary/who gets to hold reviews
```
Comment (lukecheseldine) [inline src/main/java/entities/User.java:109]: User holds Reviews which have been written about them

File: src/main/java/entities/User.java [hunk 2]
```diff
@@ -97,12 +98,18 @@ public void addCart() {
...
+    public void addReview(Review review){
+        this.reviews.add(review);
+    }
 //    public int calculateRating() {
```
Comment (lukecheseldine) [inline src/main/java/entities/User.java:113]: loop over reviews and calculate average? will need getters for this

**Review by lukecheseldine — CHANGES_REQUESTED** [review]
really really really good. i don't understand clean architecture, so i can't comment on that. i left a comment about getters on the User class. also, reformat code pls. overall this is really really really good 😄

File: src/main/java/entities/User.java (hunk 2 above)
Comment (claregillis) [reply src/main/java/entities/User.java]: just added the calculateRating feature, is that what you meant?

File: src/main/java/entities/User.java [hunk 3]
```diff
@@ -95,24 +96,54 @@ public void createListing() {
...
+     * @return the average rating of all this User's reviews
+     */
+    public int calculateRating() {
+        double rating = 0;
+        for (Review review : reviews) {
+            rating += review.getRating();
+        }
+        rating /= reviews.size();
+        return (int) Math.round(rating);
+    }
```
Comment (lukecheseldine) [inline src/main/java/entities/User.java:146]: i feel like a review could be a double, and we should allow decimal ratings. maybe round to a tenth, but not to a whole number. implementation detail tho, so i'm gonna approve for now!

**Review by lukecheseldine — DISMISSED** [review]
amazing!

**Review by danielqiuu — APPROVED** [review]

**Review by liu-samuel — DISMISSED** [review]
lgtm

**Review by helenastra — APPROVED** [review]

**Review by danielqiuu — APPROVED** [review]

**Review by ethanrstark — APPROVED** [review]

**Pull request merged**